.github/workflows/code-linter.yml

```
@@ -5,25 +5,44 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.8]
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        python-version: [3.11]
+    env:
+      POETRY_VIRTUALENVS_CREATE: false
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Display Python version
-      run: python -c "import sys; print(sys.version)"
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install flake8 flake8-docstrings yapf
-    - name: Run flake8
-      run: flake8 ./src
-    - name: Run yapf
-      run: |
-        CHECK=$(yapf --diff --recursive ./src | grep ".*(reformatted)" | wc -l)
-        if (($CHECK>0)); then exit 1; else exit 0; fi;
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install Poetry
+        run: |
+              python3 -m pip install --upgrade pip
+              pip install poetry
+      - name: Display Poetry version
+        run: |
+              poetry --version
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
+      - name: Install dependencies
+        run: |
+          poetry install
+      - name: Run YAPF
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+            cd $GITHUB_WORKSPACE
+            poetry run yapf -i -r .
+      - name: Run YAPF
+        if: matrix.os == 'macos-latest'
+        run: |
+              cd $GITHUB_WORKSPACE
+              poetry run yapf -i -r .
+      - name: Run YAPF
+        if: matrix.os == 'windows-latest'
+        run: |
+              cd %GITHUB_WORKSPACE%
+              poetry run yapf -i -r .
+        shell: cmd
```

```suggestion
      - name: Run YAPF
        run: poetry run yapf -i -r .
```

why a simple version like that doesn't work ?

---

it does work, done!